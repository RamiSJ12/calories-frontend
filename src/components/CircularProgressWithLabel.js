import React from 'react';
import PropTypes from 'prop-types';
import CircularProgress from '@material-ui/core/CircularProgress';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import commonStyles from '../utils/commonStyles';

const CircularProgressWithLabel = ({ value }) => {
  const commonClasses = commonStyles();
  return (
    <Box position="relative" display="inline-flex">
      <CircularProgress variant="determinate" value={value} size={60} />
      <Box
        top={0}
        left={0}
        bottom={0}
        right={0}
        position="absolute"
        display="flex"
        alignItems="center"
        justifyContent="center"
      >
        <Typography className={commonClasses.fontWeightBold} component="div" color="textSecondary">
          {`${value}G`}
        </Typography>
      </Box>
    </Box>
  );
};

CircularProgressWithLabel.propTypes = {
  value: PropTypes.number.isRequired,
};

export default CircularProgressWithLabel;
