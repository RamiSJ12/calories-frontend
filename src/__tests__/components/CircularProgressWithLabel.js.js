import renderer from 'react-test-renderer';
import CircularProgressWithLabel from '../../components/CircularProgressWithLabel';

describe('CircularProgressWithLabel', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<CircularProgressWithLabel />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
