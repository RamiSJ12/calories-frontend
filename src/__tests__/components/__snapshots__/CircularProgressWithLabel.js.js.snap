// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularProgressWithLabel should match snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-24"
>
  <div
    aria-valuenow={75}
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
    role="progressbar"
    style={
      Object {
        "height": 60,
        "transform": "rotate(-90deg)",
        "width": 60,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={
          Object {
            "strokeDasharray": "126.920",
            "strokeDashoffset": "31.730px",
          }
        }
      />
    </svg>
  </div>
  <div
    className="MuiBox-root MuiBox-root-25"
  >
    <div
      className="MuiTypography-root makeStyles-fontWeightBold-16 MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      75
    </div>
  </div>
</div>
`;
